### PR TITLE
Make readmes required for tests

### PR DESCRIPTION
**Pull request by yuvipanda**
And add some

### CODE REVIEW

**Comment by choldgraf** [comment]
ah shit, I just realized that you added these :-/

**Comment by choldgraf** [comment]
wanna take a look at my PR and pick which version of each you like more, and I can remove as necessary?

**Comment by yuvipanda** [comment]
@choldgraf I rebased, and only added the commit that makes READMEs mandatory :D All my text was shit anyway :D

File: tests/conftest.py [hunk 1]
```diff
@@ -28,6 +29,9 @@ def __init__(self, name, parent, path):
         self.path = path
 
     def runtest(self):
+        readme_path = os.path.join(self.path.dirname, 'README.rst')
```
Comment (minrk) [inline tests/conftest.py]: Probably best to require `README*`, not .rst. People don't tend to write .rst READMEs these days:

```python
readmes = glob.glob(os.path.join(self.path.dirname, 'README*'))
if not readmes:
    raise ...

Comment (choldgraf) [inline tests/conftest.py]: though the readme *will* need to actually be in rST syntax. We're using them to generate some documentation from these tests (for now anyway)

http://repo2docker.readthedocs.io/en/latest/

Comment (minrk) [reply tests/conftest.py]: Fair enough. Though sphinx also works with .md files via [recommonmark](https://github.com/jupyterhub/jupyterhub/blob/0.7.2/docs/source/conf.py#L55). I'm always hesitant to make people write new .rst documents.

**Pull request closed without merging**